config/base/kustomization.yaml

```diff
@@ -124,6 +124,39 @@ replacements:
           name: odh-model-controller
         fieldPaths:
           - spec.template.spec.containers.0.image
+  - source:
+      kind: ConfigMap
+      version: v1
+      name: odh-model-controller-parameters
+      fieldPath: data.nim-state
+    targets:
+      - select:
+          kind: Deployment
+          name: odh-model-controller
+        fieldPaths:
+          - spec.template.spec.containers.0.env.[name=NIM_STATE].value
```

Comment:
May you also here use `containers.[name=manager]` instead of `containers.0`? Just to prevent unexpected results if the order ever changes.